### PR TITLE
Addon Manager: Correct run_interruptable_subprocess

### PR DESCRIPTION
**Pull request by chennes**
Two refactorings and a bug fix:

1) `communicate()` has to be called after a final `kill()` to get the output. Also adds unit tests for the function. Related to #18380.
2) Renames and starts a refactoring/testing pass on the Python dependencies GUI. This needs considerably more refactoring, but this is a start.

### CODE REVIEW

**Comment by chennes** [comment]
@yorikvanhavre can you give this a quick look and merge if OK?

File: src/Mod/AddonManager/addonmanager_python_deps_gui.py [hunk 1]
```diff
@@ -441,21 +461,21 @@ def _current_python_version_is_new(cls) -> bool:
             return True
         return False
 
-    def _load_old_package_list(self) -> List[str]:
+    def _load_old_package_list(self) -> KeysView[str]:
         """Gets the list of packages from the package installation manifest"""
```
Comment (hyarion) [inline src/Mod/AddonManager/addonmanager_python_deps_gui.py:465]: There's now a mismatch between docstring+function name and return type.

I don't think it makes sense to return a `KeysView[str]`.
We should either:
* change the return to `return list(packages.keys())`, or
* refactor the function, including name docstring, to return an `Iterable[str]`. <-- I would prefer that

File: src/Mod/AddonManager/addonmanager_utilities.py [hunk 2]
```diff
@@ -417,14 +418,25 @@ def run_interruptable_subprocess(args) -> subprocess.CompletedProcess:
...
+            if (
+                hasattr(QtCore, "QThread")
+                and QtCore.QThread.currentThread().isInterruptionRequested()
+            ):
                 p.kill()
                 raise ProcessInterrupted()
+            if counter >= timeout_secs:  # The real timeout
+                p.kill()
+                stdout, stderr = p.communicate()
+                return_code = -1
```
Comment (hyarion) [inline src/Mod/AddonManager/addonmanager_utilities.py:439]: The code could be simplified by removing the counter and just look at the elapsed time instead.
```python
if time.time() - start_time >= timeout_secs
```

**Comment by chennes** [comment]
Thanks for looking at this, @hyarion -- I made your suggested fixes. Using `time.time()` made the testing a little trickier, but I think I figured something out.

**Pull request merged**